lusidtools/cocoon/instruments.py

```diff
@@ -335,50 +347,29 @@ def get_unique_identifiers(api_factory: lusid.utilities.ApiClientFactory):
                 continue
             elif len(single_search[0].external_instruments) > 0:
                 name = single_search[0].external_instruments[0].name
+                figi = single_search[0].external_instruments[0].identifiers["Figi"].value
                 break
         names.append(name)
+        figis.append(figi)
 
     enriched_column_name = "LUSID.Name.Enriched"
+    enriched_column_figi = "LUSID.Figi.Enriched"
 
     data_frame[enriched_column_name] = names
+    data_frame[enriched_column_figi] = figis
 
     # Missing mapping for name altogether
     if "name" not in list(mapping_required.keys()):
         mapping_required["name"] = enriched_column_name
+    else:
+        enrich_existing_column(data_frame, mapping_required, "name", enriched_column_name, constant_prefix)
```

Comment:
As per comment in utilities, let's explicitly return the data_frame and mapping_required or instrument_identifier_mapping from these calls so as to not rely on them being modified in place by the function.

Reply:
returns mapping as mapping_required